src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectDumpActionTests.cs

```diff
@@ -0,0 +1,112 @@
+        private readonly EndpointUtilities _endpointUtilities;
+
+        public CollectDumpActionTests(ITestOutputHelper outputHelper)
+        {
+            _outputHelper = outputHelper;
+            _endpointUtilities = new(_outputHelper);
+        }
+
+        [Theory]
+        [InlineData(DumpType.Full)]
+        [InlineData(DumpType.Mini)]
+        [InlineData(DumpType.Triage)]
+        [InlineData(DumpType.WithHeap)]
+        [InlineData(null)]
+        public async Task CollectDumpAction_Success(DumpType? dumpType)
+        {
+            DirectoryInfo uniqueEgressDirectory = null;
+
+            try {
+                DumpType? ExpectedDumpType = dumpType;
+
+                uniqueEgressDirectory = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), TempEgressDirectory, Guid.NewGuid().ToString()));
+
+                await TestHostHelper.CreateCollectionRulesHost(_outputHelper, rootOptions =>
+                {
+                    rootOptions.AddFileSystemEgress(ExpectedEgressProvider, uniqueEgressDirectory.FullName);
+
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .AddCollectDumpAction(ExpectedEgressProvider, ExpectedDumpType)
+                        .SetStartupTrigger();
+                }, async host =>
+                {
+                    IOptionsMonitor<CollectionRuleOptions> ruleOptionsMonitor = host.Services.GetService<IOptionsMonitor<CollectionRuleOptions>>();
+                    CollectDumpOptions options = (CollectDumpOptions)ruleOptionsMonitor.Get(DefaultRuleName).Actions[0].Settings;
+
+                    ICollectionRuleActionProxy action;
+                    host.Services.GetService<ICollectionRuleActionOperations>().TryCreateAction(KnownCollectionRuleActions.CollectDump, out action);
+
+                    EndpointInfoSourceCallback callback = new(_outputHelper);
+                    await using var source = _endpointUtilities.CreateServerSource(out string transportName, callback);
+                    source.Start();
+
+                    AppRunner runner = _endpointUtilities.CreateAppRunner(transportName, TargetFrameworkMoniker.Net60); // Arbitrarily chose Net60; should we test against other frameworks?
+
+                    Task<IEndpointInfo> newEndpointInfoTask = callback.WaitForNewEndpointInfoAsync(runner, CommonTestTimeouts.StartProcess);
+
+                    await runner.ExecuteAsync(async () =>
+                    {
+                        IEndpointInfo endpointInfo = await newEndpointInfoTask;
+
+                        using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(CommonTestTimeouts.DumpTimeout);
+                        CollectionRuleActionResult result = await action.ExecuteAsync(options, endpointInfo, cancellationTokenSource.Token);
+
+                        Assert.NotNull(result.OutputValues);
+                        Assert.True(result.OutputValues.TryGetValue("EgressPath", out string egressPath));
```

Comment:
I can do this for now since this is just for Dump, but seeing as most of our other actions are also going to be returning a result that uses "EgressPath" as a key, I feel like this should live somewhere that's agnostic of any specific action.